### PR TITLE
Create a BillingWrapper that ensures we are connected.

### PR DESCRIPTION
**Pull request by PEConn**
This PR adds the `ConnectedBillingWrapper` that takes a `BillingWrapper` and ensures that connect is called (and completes) before calling any of its methods.

### CODE REVIEW

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/ConnectedBillingWrapper.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
+package com.google.androidbrowserhelper.playbilling.digitalgoods;
```
Comment (andreban) [inline playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/ConnectedBillingWrapper.java:1]: nit: Missing license header.

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/ConnectedBillingWrapper.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+ */
+public class ConnectedBillingWrapper implements BillingWrapper {
+    private final BillingWrapper mInner;
+
+    private static final int NOT_CONNECTED = 0;
+    private static final int CONNECTING = 1;
+    private static final int CONNECTED = 2;
+    private int mState = NOT_CONNECTED;
+
+    private interface Callback {
```
Comment (andreban) [inline playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/ConnectedBillingWrapper.java:28]: This is more of an observation: this is the same as the `Runnable` interface. Would it make sense to use it instead (or extend if we want to add extra methods)?

**Review by andreban — APPROVED** [review]
LGTM, with the missing license header.

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/ConnectedBillingWrapper.java (hunk 2 above)
Comment (PEConn) [reply playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/ConnectedBillingWrapper.java:28]: Oh good spot, thanks!

**Pull request merged**